models/Document.php

```diff
@@ -1110,6 +1112,8 @@ public function setParentId($parentId)
     {
         $this->parentId = (int) $parentId;
         $this->parent = null;
+        $this->siblings = [];
```

Comment:
Can you explain this. The real parent doesn't change until save is called.

Reply:
when you change `$parentId` then you should be able to fetch siblings based on new `$parentId` not cached one. please see https://github.com/pimcore/pimcore/blob/7111e7423bf43926a3620e43680b1a00204f42c1/models/Document.php#L723